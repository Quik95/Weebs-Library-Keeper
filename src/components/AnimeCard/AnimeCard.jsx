import React from "react";
import PropTypes from "prop-types";
import { useStoreState, useStoreActions } from "easy-peasy";

// MUI
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import CardMedia from "@material-ui/core/CardMedia";
import Grid from "@material-ui/core/Grid";
import Typography from "@material-ui/core/Typography";

// styles
import useStyles from "./AnimeCard.styles";

// Custom components
import WatchingProgress from "./WatchingProgress/WatchingProgress";
import EditAnimeButton from "./EditAnimeButton/EditAnimeButton";
import AiringBadge from "./AiringBadge/AiringBadge";

function AnimeCard({ _id }) {
  const { title, thumbnailUrl } = useStoreState(state =>
    state.animeList.list.find(anime => anime._id === _id)
  );
  const showDialog = useStoreActions(state => state.dialog.showDialog);

  const classes = useStyles();

  return (
    <>
      <Grid item xs={12} sm={6} md={4} lg={3} xl={2} data-testid="AnimeCard">
        <AiringBadge airingIn={1424123}>
          <Card className={classes.fabFix}>
            <CardMedia image={thumbnailUrl} className={classes.animeMedia} />
            <CardContent className={classes.cardContent}>
              <Typography variant="body1" align="center" noWrap gutterBottom>
                {title}
              </Typography>
              <WatchingProgress />
            </CardContent>
            <EditAnimeButton handleOpenDialog={showDialog} />
          </Card>
        </AiringBadge>
      </Grid>
    </>
  );
}

AnimeCard.propTypes = {
  _id: PropTypes.string.isRequired
};

export default AnimeCard;
