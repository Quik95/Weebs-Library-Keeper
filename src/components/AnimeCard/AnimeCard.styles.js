import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(theme => ({
  animeMedia: {
    height: '350px',
  },
  cardContent: {
    padding: theme.spacing(1, 2),
  },
  fabFix: { overflow: 'visible', position: 'relative' },
}));
