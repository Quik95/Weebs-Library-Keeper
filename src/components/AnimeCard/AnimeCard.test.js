import React from 'react';
import { render } from '@testing-library/react';

import AnimeCard from './AnimeCard';

describe('<AnimeCard/>', () => {
  const { getByTestId } = render(<AnimeCard />);

  it('Renders correctly', () => {
    const wrapper = getByTestId('AnimeCard');
    expect(wrapper).toMatchSnapshot();
  });
});
