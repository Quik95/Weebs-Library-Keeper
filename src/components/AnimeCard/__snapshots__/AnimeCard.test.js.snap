// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AnimeCard/> Renders correctly 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Grid))
    data-testid="AnimeCard"
    item={true}
    xs={3}
  >
    <WithStyles(ForwardRef(Card))
      className="makeStyles-fabFix-3"
    >
      <WithStyles(ForwardRef(CardMedia))
        className="makeStyles-animeMedia-1"
        image="https://s4.anilist.co/file/anilistcdn/media/anime/cover/large/bx105333-5p1MKBlGxZFF.jpg"
      />
      <WithStyles(ForwardRef(CardContent))
        className="makeStyles-cardContent-2"
      >
        <WithStyles(ForwardRef(Typography))
          align="center"
          gutterBottom={true}
          noWrap={true}
          variant="body1"
        >
          Dr. Stone
        </WithStyles(ForwardRef(Typography))>
        <Memo(WatchingProgress) />
      </WithStyles(ForwardRef(CardContent))>
      <Memo(EditAnimeButton)
        handleOpenDialog={[Function]}
      />
    </WithStyles(ForwardRef(Card))>
  </WithStyles(ForwardRef(Grid))>
  <EditAnimeDialog
    handleClose={[Function]}
    maxWidth="md"
    open={false}
  />
</Fragment>
`;
